zenoh-jni/src/key_expr.rs

```diff
@@ -152,3 +115,74 @@ pub(crate) unsafe extern "C" fn Java_io_zenoh_jni_JNIKeyExpr_freePtrViaJNI(
 ) {
     Arc::from_raw(ptr);
 }
+
+pub(crate) fn decode_key_expr(env: &mut JNIEnv, key_expr: &JString) -> Result<KeyExpr<'static>> {
+    let key_expr_str = decode_string(env, key_expr).map_err(|err| {
+        Error::Jni(format!(
+            "Unable to get key expression string value: {}",
+            err
+        ))
+    })?;
+    let key_expr = KeyExpr::try_from(key_expr_str).map_err(|err| {
+        Error::Jni(format!(
+            "Unable to create key expression from string: {}",
+            err
+        ))
+    })?;
+    Ok(key_expr)
+}
+
+pub(crate) fn autocanonize_key_expr(
+    env: &mut JNIEnv,
+    key_expr: &JString,
+) -> Result<KeyExpr<'static>> {
+    let key_expr_str = decode_string(env, key_expr).map_err(|err| {
+        Error::Jni(format!(
+            "Unable to get key expression string value: {}",
+            err
+        ))
```

Comment:
nit: Potential to statically define these errors somewhere, and send back references to the errors. May change performance, reviewer has not tried, up to OP to if OP wants to try.

Reply:
It's a pending task (although low priority) to refactor the error handling code on zenoh-jni. I will consider this for the future.